### PR TITLE
Changed yii.confirm context bind

### PR DESCRIPTION
**Pull request by lichunqiang**
Related #10097

### CODE REVIEW

**Comment by SilverFire** [comment]
Looks good. Need to add a changelog line

File: framework/assets/yii.js [hunk 1]
```diff
@@ -329,7 +329,7 @@ yii = (function ($) {
             }
 
             if (message !== undefined) {
-                pub.confirm(message, function () {
+                $.proxy(pub.confirm, this)(message, function () {
                     pub.handleAction($this, event);
```
Comment (samdark) [inline framework/assets/yii.js]: Couldn't `$this` be used as `this` now?

Comment (samdark) [inline framework/assets/yii.js]: Also it seems it's not the only method like that. There's `handleAction` in `else` branch. Should it be applied there as well?

Comment (lichunqiang) [inline framework/assets/yii.js]: It's just change the `yii.confirm` function context bind.
```

Comment (lichunqiang) [inline framework/assets/yii.js]: if use `$this` in the `yii.confirm` function, we no need to wrap `this` and change to jquery dom object.

Comment (SilverFire) [inline framework/assets/yii.js]: @lichunqiang you show the closure signature `function(message, ok, cancel)`, but I can't figure out  where the `cancel` attribute will be passed from.

Comment (lichunqiang) [inline framework/assets/yii.js]: @SilverFire 
```
if (message !== undefined) {
	pub.confirm(message, function () {
		pub.handleAction($this, event);
	});
} else {
	pub.handleAction($this, event);
}
```

this is the original code.

Comment (SilverFire) [inline framework/assets/yii.js]: @lichunqiang I'm taking about your example of `yii.confirm` overriding 2 posts before. Okay, it's fine

**Comment by SilverFire** [comment]
I like this change and the code works good. As it it a BC-breaking change, we should write about it in `UPGRADE.md`. Also please add a line to `CHANGELOG.md`.

**Comment by SilverFire** [comment]
May be merged

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ Yii Framework 2 Change Log
 - Chg #9411: `DetailView` now automatically sets container tag ID in case it's not specified (samdark)
 - Chg #9953: `TimestampBehavior::getValue()` changed to make value processing consistent with `AttributeBehavior::getValue()` (silverfire)
 - New #10083: Added wrapper for PHP webserver (samdark)
+- Enh #10098: Changed `yii.confirm` context bind to triggered dom element. (lichunqiang)
```
Comment (samdark) [inline framework/CHANGELOG.md]: Should be reordered before merge.

Comment (samdark) [inline framework/CHANGELOG.md]: dom → DOM

Comment (samdark) [inline framework/CHANGELOG.md]: No need for trailing dot.

File: framework/UPGRADE.md [hunk 3]
```diff
@@ -25,7 +25,8 @@ initialization to support wider range of allowed characters. Because of this cha
   See the [Cache Flushing Guide](http://www.yiiframework.com/doc-2.0/guide-caching-data.html#cache-flushing)
   - If you implement `parseRequest()` or `createUrl()` and rely on parameter names, call `substitutePlaceholderNames()`
   in order to replace temporary IDs with parameter names after doing matching.
-
+* The context of `yii.confirm` function in `yii.js` was changed to the triggered dom element.
```
Comment (samdark) [inline framework/UPGRADE.md]: dom → DOM

Comment (samdark) [inline framework/UPGRADE.md]: DOM element can't be triggered. It's an event that is triggered.

File: framework/UPGRADE.md [hunk 4]
```diff
@@ -25,7 +25,8 @@ initialization to support wider range of allowed characters. Because of this cha
...
+  - If you rewrited the `yii.confirm` function, you can get the triggered dom element as: `this` or `$(this)` to get jquery object.
```
Comment (samdark) [inline framework/UPGRADE.md]: rewrited → overrode

Comment (samdark) [inline framework/UPGRADE.md]: dom → DOM

Comment (samdark) [inline framework/UPGRADE.md]: Again, event is triggered, not DOM-element.

Comment (samdark) [inline framework/UPGRADE.md]: jquery → jQuery

**Comment by samdark** [comment]
Except minor wording things it looks fine to me.

**Comment by lichunqiang** [comment]
my bad english

**Comment by lichunqiang** [comment]
```
* The context of `yii.confirm` function in `yii.js` was changed to the DOM element which is the triggered event target.
  - If you overrode the `yii.confirm` function, you can get the DOM element as: `this` or `$(this)` to get jQuery object.
```
CHANGELOG:
```
- Enh #10098: Changed `yii.confirm` context bind to the event's target DOM element which is triggered by clickable or changeable elements (lichunqiang)
```

@SilverFire Thanks for a review....

**Comment by SilverFire** [comment]
Merged with minor adjustments https://github.com/yiisoft/yii2/commit/1a48d248301bcf30f3d20fb17ac067db5a2efc11

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**